### ISSUE TITLE
Question about the sampling strategies

### ISSUE THREAD

**Issue by Hugo-cell111** [WangChangqi98/CSS#6]
Hi! In your paper three kinds of sampling strategies are introduced and the prediction in representation space is considered. But in the final version of your [code](https://github.com/WangChangqi98/CSS/blob/main/generalframeworks/loss/loss.py#L99), the prediction in the logits space is considered instead. Have you tried the sampling strategies as said in your paper? I might miss some important details. Thanks!

**Comment by WangChangqi98** [comment]
Hi, I'm sorry for confusing you, do you mean valid sampling strategy? The weak threshold in valid sampling strategy is used for sampling reliable pseudo-labels from the representation space. So the indicator from representation space is only used in cross pseudo-labelling strategy and specifically, in its unsupervised loss. The pseudo-labels from logit space is used to supervised the contrastive learning process in cross pseudo-labelling strategy, we use the confidence as the indicator since it's corresponding to the reliability of pseudo-labels. In mix pseudo-labelling strategy, we use the pseudo-labels from the logit and the representation space in a mix way, and here we only use the confidence as the indicator to further polish the pseudo-labels. Am I making sense?

**Comment by Hugo-cell111** [comment]
Yes! So you mean sampling the negative keys in contrastive loss is guided by the pseudo labels in logits space, not in representation space, do you? And thank you for your exhaustive explanation!

**Comment by WangChangqi98** [comment]
yes, you are welcome and thank for focusing on our work.

**Comment by Hugo-cell111** [comment]
Thanks!

**Comment by Hugo-cell111** [comment]
Hi! Sorry to bother you again. Have you tried not using the threshold to filter the unrealiable prediction in mix-label strategy? And does huge performance gap exist between using or not using the threshold in the [code](https://github.com/WangChangqi98/CSS/blob/8aec4a874c94dfba76c5150a89882ebe055e6552/generalframeworks/loss/loss.py#L48)? Thanks!

**Comment by WangChangqi98** [comment]
Yes I have tried it, and the performance is poor. The sampling strategy is very important. Similar conclusion can be seen in ReCo.

**Comment by Hugo-cell111** [comment]
Thanks a lot, I will consider the threshold later. Thank you!
